src/proxy.rs

```diff
@@ -107,17 +109,23 @@ pub fn filter_frontend_request_headers(headers: &Headers) -> Headers {
 ///
 /// The primary purpose of this function is to add and remove headers as required by an
 /// intermediary conforming to the HTTP spec.
-fn map_request(req: server::Request, url: &str) -> client::Request {
-    let mut r = client::Request::new(req.method().clone(), url.parse().unwrap());
-
+fn map_request(req: server::Request, url: Url) -> client::Request {
     let via = create_via_header(
         req.headers().get::<Via>(),
         req.version());
 
     let mut headers = filter_frontend_request_headers(req.headers());
     headers.set(via);
-    r.headers_mut().extend(headers.iter());
 
+    // add host header related to backend
```

Comment:
Why is the host header changing? If I am sending a request to www.example.com, then the `Host` header should remain www.example.com even if weldr passing the request to a backend that is using a hostname.